datatransfer/impl/graphsync/graphsync_whitebox_test.go

```diff
@@ -1,4 +1,5 @@
 package graphsyncimpl
```

Comment:
I just noticed this is not in the test package. it should be. the package should be graphsyncimpl_test. This should remove the need to define it twice.

Reply:
While it does test an exported function, the test is verifying the logic by looking unexported impl fields.

Reply:
yea, I'm not sure when that changed, but we should be testing if the subscriptions get called, which is the behavior, not whether the subscriber count which is an internal variable is correct.

I think the original version used channels, but probably you don't need them -- you could just use call counts.